rclcpp/include/rclcpp/create_publisher.hpp

```diff
@@ -29,6 +29,46 @@
 namespace rclcpp
 {
 
+/// Create and return a publisher of the given MessageT type.
+/**
+ * The NodeT type only needs to have a method called get_node_topics_interface()
+ * which returns a shared_ptr to a NodeTopicsInterface.
+ */
+template<
+  typename MessageT,
+  typename AllocatorT = std::allocator<void>,
+  typename PublisherT = rclcpp::Publisher<MessageT, AllocatorT>,
+  typename NodeT>
+std::shared_ptr<PublisherT>
+create_generic_publisher(
```

Comment:
given that the return type is just a regular publisher, I would advise against renaming the function to `generic` and have it part of the overloads as originally designed.